ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -998,12 +998,12 @@ getDocsBatch
   :: HscEnv
   -> Module  -- ^ a moudle where the names are in scope
   -> [Name]
-  -> IO (Map.Map Name (Either T.Text (Maybe HsDocString, Map.Map Int HsDocString)))
+  -> IO (Either ErrorMessages (Map.Map Name (Either T.Text (Maybe HsDocString, Map.Map Int HsDocString))))
 getDocsBatch hsc_env _mod _names = do
     ((_warns,errs), res) <- initTc hsc_env HsSrcFile False _mod fakeSpan $ Map.fromList <$> traverse findNameInfo _names
     case res of
-        Just x  -> return $ fun x
-        Nothing -> throwErrors errs
```

Comment:
Stopping throwing errors into IO.

I basically want to surface stuff that was hidden into `IO` &, to have guarantees that transformations I do further with the code are isomorphic and safe and does not change the behavior even to what goes into exceptions.